Add codespell support (config, workflow to detect/not fix) and make it fix few typos

yarikoptic opened the pull request:
codespell is not new to this project but not really part of the automated workflow(s) (precommit, CI).

There is `Makefile` rule but had hardcoded settings into it, so people had to use `make` and could not `pre-commit` or directly -- I moved configuration into `pyproject.yaml` so now if you have codespell (and tomli) installed - you can just codespell. 

CI workflow has 'permissions' set only to 'read' so also should be safe.

Also added pre-commit hook for it so no new typos would sneak in.

attn @rhatdan as I believe the codespell user ;)

Luap99 commented on .github/workflows/codespell.yml:
We do not use github actions for the most part so please delete that.

If we want to enforce this on CI it must run in cirrus as part of make validate-sources, given you already hooked this into pre-commit this is already the case so it works

yarikoptic replied:
I will
- delete GH action
- add a TEMP commit with a typo to ensure they do get detected with pre-commit (will take PR into draft for the duration of TEMP commit presence).

yarikoptic replied:
Apparently both "Cirrus CI / Windows Cross" and "Cirrus CI / Validate source code changes" ran pre-commit and failed on `codespell`. As the point of test catching new typos is fulfilled, I will remove TEMP commit now and take out of draft.

Luap99 commented on pyproject.toml:
```diff
@@ -13,3 +13,10 @@ exclude = '''
 )/
 '''
+
+[tool.codespell]
```
I don't like that, it feels totally arbitrary to put this unrelated configuration in a python config file.

I recommend to use .codespellrc

yarikoptic replied:
Up to you.  I am even a little on your side since then it would not require optional `tomli`.

Just 1 more chance to reconsider under the following arguments

- often people prefer avoiding pollution of root directory with per-tool configuration files.
- [you already have the one (and only content) `tool.black` in pyproject.toml](https://github.com/containers/podman/blob/main/pyproject.toml)
  - FWIW codespell is written in Python , if argument "black is written in Python" comes up ;-)

So let me know

Luap99 replied:
I don't want this in a random python file, we are not a python project so please remove it from there. I don't think anyone will ever look into this file ever and fine the codespell config there.

Luap99 reviewed (changes requested):
Thanks, I am fine with having codespell in CI but you properly want to wait for feedback from other maintainers first before spending more time on this in case others do not like it

rhatdan commented:
I like this alot, thanks @yarikoptic. Fix the couple of issues @Luap99 found, and I will give it a LGTM

Luap99 commented:
Also if you repush can you please fix up the commit messages for the `[DATALAD RUNCMD]` commits, that is completely  useless data for any human reader of the commit messages

rhatdan commented:
I would be fine with just merging the commits into one PR.

yarikoptic commented:
> Also if you repush can you please fix up the commit messages for the `[DATALAD RUNCMD]` commits, that is completely useless data for any human reader of the commit messages

FWIW: I can easily adjust short commit message and can remove internal record on how the modification was made (some humans are curious creatures and actually do like to see command line invocation which resulted in the committed diff... More on `datalad run` at https://handbook.datalad.org/en/latest/basics/basics-run.html ... It also becomes "fun" to do "functional rebase" whenever commit is taken not as a diff, but rather as "do this action" (like `codespell -w` or some `sed -i` invocation), thus avoiding dealing with conflicts or simply missing some new changes to be done.

With that in mind, just tell me final verdict of

- C1: leave along as is
- C2: adjust short summary
- C3: adjust full commit message and remove embedded json record with the command

rhatdan commented:
LGTM
@Luap99 PTAL

Luap99 commented:
> C3: adjust full commit message and remove embedded json record with the command

I am reading commit message and this random stuff just wastes every readers time trying to make sense of this and why it is there. Please just squash the codespell fixes in a commit and name it something like fix issues reported by codespell or something like this.

Luap99 commented on Makefile:
```diff
@@ -309,7 +309,7 @@ test/version/version: version/version.go
 .PHONY: codespell
 codespell:
-	codespell -S bin,vendor,.git,go.sum,.cirrus.yml,"*.fish,RELEASE_NOTES.md,*.xz,*.gz,*.ps1,*.tar,swagger.yaml,*.tgz,bin2img,*ico,*.png,*.1,*.5,copyimg,*.orig,apidoc.go" -L ddress,secon,passt,bu,hastable,te,clos,ans,pullrequest,uint,iff,od,seeked,splitted,marge,erro,hist,ether,specif -w
+	codespell -w
```
it would help to add a comment here saying where the new config is for authors used to edit the rules here inline.

yarikoptic replied:
added a comment

yarikoptic commented:
- moved config into new `.codespellrc`
- rebased and reran fixing  (there was one new typo)
- squashed multiple commits and minimized their commit messages - now only 3 commits.  Feel welcome to squash into a single one upon merging

rhatdan commented:
@yarikoptic still working on this?  Looks like all it needs is to adjust the commit message.

yarikoptic commented:
Well, I thought I was done with it! ;)  now spotted that failed CI apparently complaining about overly long subject line!

```
 * 9bde6b551 "Add codespell config, pre-commit definition, and move used options from Makefile into config" ... FAIL
  - FAIL - commit subject exceeds 90 characters
 ```

ok,
-  reworded
-  Since that rewrote history -- rebased again
- saw that 2 new typos detected, made it fix those and fixup that last commit with other fixes
- force pushed

rhatdan commented:
/approve
/lgtm

rhatdan commented:
Thanks @yarikoptic
